src/mappings/poktroll/reports.ts

```diff
@@ -0,0 +1,314 @@
+    {unstakingApps, unstakingTokensByApp},
+    {unstakedApps, unstakedTokensByApp},
+    {stakedGateways, stakedTokensByGateway},
+    {unstakedGateways, unstakedTokensByGateway},
+    blockFromDB
+  ] = await Promise.all([
+    getRelaysData(block),
+    getTransactionsData(block),
+    getStakedSuppliersData(),
+    getUnstakingSuppliersData(),
+    getTook(block),
+    getUnstakedSuppliersData(block.block.id),
+    getStakedAppsData(),
+    getUnstakingAppsData(),
+    getUnstakedAppssData(block.block.id),
+    getStakedGatewaysData(),
+    getUnstakedGatewaysData(block.block.id),
+    Block.get(block.block.id)!
+  ] as const)
+
+  const blockEntity = blockFromDB as Block
+
+  blockEntity.totalComputedUnits = computedUnits
+  blockEntity.totalRelays = relays
+  blockEntity.failedTxs = invalidTxs
+  blockEntity.validTxs = validTxs
+  blockEntity.took = took
+  blockEntity.totalTxs = validTxs + invalidTxs
+  blockEntity.stakedSuppliers = stakedSuppliers
+  blockEntity.suppliersStakedTokens = stakedTokensBySupplier
+  blockEntity.unstakingSuppliers = unstakingSuppliers
+  blockEntity.suppliersUnstakingTokens = unstakingTokensBySupplier
+  blockEntity.unstakedSuppliers = unstakedSuppliers
+  blockEntity.unstakedTokensOfSuppliers = unstakedTokensBySupplier
+  blockEntity.stakedApps = stakedApps
+  blockEntity.appsStakedTokens = stakedTokensByApp
+  blockEntity.unstakingApps = unstakingApps
+  blockEntity.appsUnstakingTokens = unstakingTokensByApp
+  blockEntity.unstakedApps = unstakedApps
+  blockEntity.unstakedTokensOfApps = unstakedTokensByApp
+  blockEntity.stakedGateways = stakedGateways
+  blockEntity.gatewaysStakedTokens = stakedTokensByGateway
+  blockEntity.unstakingGateways = unstakedGateways
+  blockEntity.gatewaysUnstakingTokens = unstakedTokensByGateway
+
+  await Promise.all([
+    blockEntity.save(),
+    store.bulkCreate('RelayByBlockAndService', relaysByService.map(relay => ({
+      id: `${block.block.id}-${relay.service}`,
+      relays: relay.relays,
+      amount: relay.amount,
+      computedUnits: relay.computedUnits,
+      blockId: block.block.id,
+      serviceId: relay.service,
+    } as RelayByBlockAndServiceProps))),
+    store.bulkCreate('StakedSuppliersByBlockAndService', stakedSuppliersByService.map(staked => ({
+      id: `${block.block.id}-${staked.service}`,
+      tokens: staked.tokens,
+      amount: staked.amount,
+      blockId: block.block.id,
+      serviceId: staked.service,
+    } as StakedSuppliersByBlockAndServiceProps))),
+    store.bulkCreate('StakedAppsByBlockAndService', stakedAppsByService.map(staked => ({
+      id: `${block.block.id}-${staked.service}`,
+      tokens: staked.tokens,
+      amount: staked.amount,
+      blockId: block.block.id,
+      serviceId: staked.service,
+    } as StakedAppsByBlockAndServiceProps))),
+  ])
+}
+
+async function getRelaysData(block: CosmosBlock){
+  const relays = await EventClaimSettled.getByFields([["blockId", "=", block.block.id]], {})
```

Comment:
I'm sure this is not a good idea, here you can have so many records, so it will need pagination.

In the other hand we maybe want this kind of summary somehow achieve by running database queries instead? (I know we can't right now with the current ORM of SubQuery, I will ask them)